pkg/provider/azure_storageaccount.go

```diff
@@ -274,6 +274,103 @@ func (az *Cloud) EnsureStorageAccount(accountOptions *AccountOptions, genAccount
+	privateDNSZone := privatedns.PrivateZone{Location: &location}
+	if err := az.privatednsclient.CreateOrUpdate(ctx, az.ResourceGroup, PrivateDNSZoneName, privateDNSZone, true); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (az *Cloud) createVNetLink(ctx context.Context, vNetLinkName string) error {
+	klog.V(5).Infof("Creating virtual link for vnet(%s) and DNS Zone(%s) in resourceGroup(%s)", vNetLinkName, PrivateDNSZoneName, az.ResourceGroup)
+	location := LocationGlobal
+	vnetID := fmt.Sprintf("/subscriptions/%s/resourceGroups/%s/providers/Microsoft.Network/virtualNetworks/%s", az.SubscriptionID, az.ResourceGroup, az.VnetName)
+	registrationEnabled := false
+	parameters := privatedns.VirtualNetworkLink{
+		Location: &location,
+		VirtualNetworkLinkProperties: &privatedns.VirtualNetworkLinkProperties{
+			VirtualNetwork:      &privatedns.SubResource{ID: &vnetID},
+			RegistrationEnabled: &registrationEnabled},
+	}
+	if err := az.virtualNetworkLinksClient.CreateOrUpdate(ctx, az.ResourceGroup, PrivateDNSZoneName, vNetLinkName, parameters, false); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (az *Cloud) createPrivateDNSZoneGroup(ctx context.Context, dnsZoneGroupName string, privateEndpointName string) error {
+	klog.V(5).Infof("Creating private DNS zone group(%s) with privateEndpoint(%s), vNetName(%s), resourceGroup(%s)", dnsZoneGroupName, privateEndpointName, az.VnetName, az.ResourceGroup)
+	privateDNSZoneID := fmt.Sprintf("/subscriptions/%s/resourceGroups/%s/providers/Microsoft.Network/privateDnsZones/%s", az.SubscriptionID, az.ResourceGroup, PrivateDNSZoneName)
+	dnsZoneName := PrivateDNSZoneName
+	privateDNSZoneConfig := network.PrivateDNSZoneConfig{
+		Name: &dnsZoneName,
+		PrivateDNSZonePropertiesFormat: &network.PrivateDNSZonePropertiesFormat{
+			PrivateDNSZoneID: &privateDNSZoneID},
+	}
+	privateDNSZoneConfigs := []network.PrivateDNSZoneConfig{privateDNSZoneConfig}
+	privateDNSZoneGroup := network.PrivateDNSZoneGroup{
+		PrivateDNSZoneGroupPropertiesFormat: &network.PrivateDNSZoneGroupPropertiesFormat{
+			PrivateDNSZoneConfigs: &privateDNSZoneConfigs,
+		},
+	}
+	if err := az.privatednszonegroupclient.CreateOrUpdate(ctx, az.ResourceGroup, privateEndpointName, dnsZoneGroupName, privateDNSZoneGroup, false); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (az *Cloud) disablePublicAccess(ctx context.Context, storageAccount *storage.Account) error {
+	klog.V(5).Infof("Disable public access to storage account(%s) in resourceGroup(%s)", *storageAccount.Name, az.ResourceGroup)
+	networkRuleSet := storageAccount.NetworkRuleSet
+	networkRuleSet.DefaultAction = storage.DefaultActionDeny
+	accountUpdateParameters := storage.AccountUpdateParameters{
+		AccountPropertiesUpdateParameters: &storage.AccountPropertiesUpdateParameters{
+			NetworkRuleSet: networkRuleSet,
+		},
+	}
+	if rerr := az.StorageAccountClient.Update(ctx, az.ResourceGroup, *storageAccount.Name, accountUpdateParameters); rerr != nil {
```

Comment:
that would require another storage account update operation, why not:
 - if create private end point:
    - set  disablePublicAccess when creating storage account

Reply:
Done